dev/com.ibm.ws.security.oidc.social_fat.backchannelLogout/fat/src/com/ibm/ws/security/oidc_social/backchannelLogout/fat/LogoutTokenValidationTests.java

```diff
@@ -0,0 +1,1472 @@
+        acrValue.put("values", testJsonArray);
+
+        HashMap<String, Object> claimMap = new HashMap<String, Object>() {
+            {
+
+                // TODO - once the runtime validation is updated, re-enable the next line and remove the last check in this method
+                //put(Constants.PAYLOAD_NOT_BEFORE_TIME_IN_SECS, 2056387597);
+                put("email", "joe@something.com");
+                put(Constants.PAYLOAD_AUTHZ_TIME_IN_SECS, 478550797);
+                put(Constants.PAYLOAD_AUTHORIZED_PARTY, "noOne");
+                put(Constants.PAYLOAD_AT_HASH, new String(Base64.encode("myDogHasFleas".getBytes())));
+                put(Constants.PAYLOAD_CLASS_REFERENCE, acrValue);
+                put(Constants.PAYLOAD_METHODS_REFERENCE, testJsonArray);
+                put(Constants.PAYLOAD_GROUP, testJsonArray);
+                put(Constants.PAYLOAD_GROUPS, testJsonArray);
+                put(Constants.PAYLOAD_USER_PRINCIPAL_NAME, "someFakeUser");
+                put(Constants.PAYLOAD_TOKEN_TYPE, Constants.TOKEN_TYPE_BEARER);
+
+            }
+        };
+
+        for (Map.Entry<String, Object> entry : claimMap.entrySet()) {
+
+            genericAddedUpdatedClaimsTest(entry.getKey(), entry.getValue());
+        }
+
+        // TODO - once the runtime validation is updated, remove the next line and restore the line in the map above
+        genericInvalidClaimTest(Constants.PAYLOAD_NOT_BEFORE_TIME_IN_SECS, 2056387597, "InvalidJwtException");
+
+    }
+
+    /*****************************************
+     * Signature tests
+     *****************************************/
+    /**
+     * The RP config specifies HS256 and the token is signed using HS256 - expect the request to succeed
+     *
+     * @throws Exception
+     */
+    @Test
+    public void LogoutTokenValidationTests_sign_HS256() throws Exception {
+        genericSignatureAlgTest("clientSignHS256", Constants.SIGALG_HS256, Constants.SIGALG_HS256);
+    }
+
+    /**
+     * The rp config specifies HS256, the token is signed using another algorithm - expect the request to fail
+     *
+     * @throws Exception
+     */
+    @Test
+    public void LogoutTokenValidationTests_sign_HS256_mismatch() throws Exception {
+        genericSignatureAlgTest("clientSignHS256", Constants.SIGALG_HS256, Constants.SIGALG_HS512);
+    }
+
+    /**
+     * The RP config specifies HS384 and the token is signed using HS384 - expect the request to succeed
+     *
+     * @throws Exception
+     */
+    @Test
+    public void LogoutTokenValidationTests_sign_HS384() throws Exception {
+        genericSignatureAlgTest("clientSignHS384", Constants.SIGALG_HS384, Constants.SIGALG_HS384);
+    }
+
+    /**
+     * The rp config specifies HS384, the token is signed using another algorithm - expect the request to fail
+     *
+     * @throws Exception
+     */
+    @Test
+    public void LogoutTokenValidationTests_sign_HS384_mismatch() throws Exception {
```

Comment:
Shouldn't just one algorithm mismatch test be sufficient? Tests for mismatches in HS256 vs. HS512, HS384 vs. HS256, and HS512 vs. HS384 are all in the same equivalence class. Same goes for all of the other signature algorithms. The HS* algorithms would be their own equivalence class, then RS* algorithms, then ES* algorithms.

Reply:
One of each is probably enough, but, I'm not really planning on testing more end to end function using all variations of signing/encrypting.  So, these tests are quick and give us some coverage.